xls/examples/hack.x

```diff
@@ -0,0 +1,256 @@
+// Copyright 2020 The XLS Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// An incomplete implementation of a the Hack CPU from the https://www.nand2tetris.org/ course.
+//
+// Implemented:
+// - ALU
+// - A/C instructions decoding
+// - RAM addressing
+// - ROM PC addressing
+// Limitation:
+// - missing screen and keyboard memory based IO
+// - smaller RAM and ROM size
+
+fn decode_a_instruction(ins: u16) -> (u15) {
+    let value: u15 = ins[0+:u15];
+    (value,)
+}
+
+#![test]
+fn decode_a_test() {
+    // @21
```

Comment:
Maybe a file level comment on what the @ comments mean?

Reply:
it's actually an instruction! (added a comment).